test_model.py

```diff
@@ -0,0 +1,179 @@
+
+    # If more than one time-series match the given metric, raise an error
+    if len(train_data) > 1:
+        _LOGGER.error("Multiple timeseries matching %s were found")
+        _LOGGER.error("The timeseries matched were: ")
+        for timeseries in train_data:
+            print(timeseries.metric_name, timeseries.label_config)
+        _LOGGER.error("One metric should be specific to a single time-series")
+        raise ValueError
+
+    # Download test data
+    test_data_list = pc.get_metric_range_data(
+        metric_name=metric,
+        start_time=rolling_data_window,
+        chunk_size=str(Configuration.retraining_interval_minutes) + "m",
+    )
+
+    _LOGGER.info("Downloaded metric data")
+
+    model_mp = model.MetricPredictor(train_data[0], rolling_data_window_size=None)
+
+    mlflow.set_experiment(train_data[0].metric_name)
+    mlflow.start_run()
+    mlflow_run_id = mlflow.active_run().info.run_id
+
+    # keep track of the model name as a mlflow run tag
+    mlflow.set_tag("model", model_mp.model_name)
+
+    # keep track of labels as tags in the mlflow experiment
+    for label in model_mp.metric.label_config:
+        mlflow.set_tag(label, train_data[0].label_config[label])
+
+    # store the metric with labels as a tag so it can be copied into grafana to view the real metric
+    mlflow.set_tag("metric", metric)
+
+    # initial run with just the train data
+    model_mp.train(train_data[0], Configuration.retraining_interval_minutes)
+
+    # store the predicted dataframe
+    predicted_df = model_mp.predicted_df
+
+    for item in range(len(test_data_list) - 1):
+        # the true values for this training period
+        true_values = Metric(test_data_list[item + 1])
+        true_values.metric_values = true_values.metric_values.set_index("ds")
+
+        # for each item in the test_data list, update the model (append new data and train it)
+        model_mp.train(test_data_list[item], len(true_values.metric_values))
+
+        # store the prediction df for every interval
+        predicted_df = predicted_df + model_mp.predicted_df
+
+        true_values.metric_values["yhat"] = model_mp.predicted_df["yhat"]
+        true_values.metric_values["yhat_upper"] = model_mp.predicted_df["yhat_upper"]
+        true_values.metric_values["yhat_lower"] = model_mp.predicted_df["yhat_lower"]
+
+        metric_timestamp = true_values.metric_values.index.values[
+            int(len(true_values.metric_values) / 2)
+        ]
+        metric_timestamp = int(metric_timestamp.astype("uint64") / 1e6)
+
+        # calculate predicted anomaly
+        model_mp.predicted_df["anomaly"] = np.where(
+            (
+                ((true_values.metric_values["y"] >= true_values.metric_values["yhat_upper"]))
+                | (true_values.metric_values["y"] <= true_values.metric_values["yhat_lower"])
+            ),
+            1,
+            0,
+        )
+
+        # Label True Anomalies
+        thresold_value = 0.001
```

Comment:
I think the `threshold_value` should be a configurable parameter so that user can tune it based on their specific test case.

Reply:
good idea

Reply:
Changed it, and some parameters are now being logged to mlflow